tkp/accessors/fitsimage.py

```diff
@@ -124,23 +124,34 @@ def parse_frequency(self):
         try:
             header = self.header
             if header['TELESCOP'] in ('LOFAR', 'AARTFAAC'):
-                freq_eff = header['RESTFRQ']
-                if 'RESTBW' in header:
-                    freq_bw = header['RESTBW']
+                if 'RESTFRQ' in header:
+                    freq_eff = header['RESTFRQ']
+                    if 'RESTBW' in header:
+                        freq_bw = header['RESTBW']
 
-                else:
-                    logger.warning("bandwidth header missing in image {},"
+                    else:
+                        logger.warning("bandwidth header missing in image {},"
                                    " setting to 1 MHz".format(self.url))
-                    freq_bw = 1e6
+                        freq_bw = 1e6
+                else:
+                    if ('CTYPE3' in header) and (header['CTYPE3'] in ('FREQ', 'VOPT')):
+                        freq_eff = header['CRVAL3']
+                        freq_bw = header['CDELT3']
+                    elif ('CTYPE4' in header) and (header['CTYPE4'] in ('FREQ', 'VOPT')):
+                        freq_eff = header['CRVAL4']
+                        freq_bw = header['CDELT4']
+                    else:
+                        freq_eff = header['RESTFREQ']
+                        freq_bw = 0.0
```

Comment:
If the bandwidth is not found here it is set to 0.0 (same below), where as above it is set to 1.0. 

Setting to 0.0 can lead to unexpected behaviour as (from memory I think this is right) it doesn't calculate the metrics even if the other measurements have the same frequency. The 1.0 default as above is safer.

Reply:
I've found that 0 works, but basically it then sets up a new frequency band for every image. So agreed this is not desirable behaviour though there is also a work around in job_params.cfg. 1 also isn't great but avoids unexpected 0 problems... So, agreed, I will set these to 1 as well.